Introduce `MessageEncoderInterface` (#308)
yiisoft/yii-queue: docs/guide/en/consuming-messages-from-external-systems.md

The key idea is simple:

- The queue adapter reads a *raw payload* (usually a string) from the broker.
- The adapter passes that payload to a `Yiisoft\Queue\Message\Serializer\MessageSerializerInterface` implementation (by default `MessageSerializer`).

> - `MessageSerializer` delegates wire encoding to a `Yiisoft\Queue\Message\Serializer\MessageEncoderInterface` implementation.

Reviewer: "wire encoding" doesn't read clearly for non-natives. I'd adjust it to be simpler.

Author: fixed

> - `MessageSerializer` assembles the message into an array and delegates encoding it to/from a string to a `Yiisoft\Queue\Message\Serializer\MessageEncoderInterface` implementation.

Reviewer: ```suggestion
- `MessageSerializer` deserializes the payload into an array. It delegates decoding of payload format to `Yiisoft\Queue\Message\Serializer\MessageEncoderInterface` implementation.
```

- By default, `yiisoft/queue` config binds `MessageEncoderInterface` to `Yiisoft\Queue\Message\Serializer\JsonMessageEncoder`.

`JsonMessageEncoder` is only the default implementation. You can replace it with your own encoder by rebinding `Yiisoft\Queue\Message\Serializer\MessageEncoderInterface` in your DI configuration.

> So, external systems should produce the **same payload format** that `MessageSerializer` expects. The payload **shape** (`type`, `data`, `meta` keys) is defined by `MessageSerializer` regardless of the encoder; the encoder only converts between the wire representation and the internal array (JSON ↔ array by default).

Reviewer: "wire representation" — same. Not clear for non-natives. Could it be simplified?

Author: fixed

> So, external systems should produce the **same payload format** that `MessageSerializer` expects. The payload **shape** (`type`, `data`, `meta` keys) is defined by `MessageSerializer` regardless of the encoder; the encoder only converts between the raw string and an associative array (JSON ↔ array by default).

Reviewer: ```suggestion
External systems should produce the **same payload format** that `MessageSerializer` expects. The payload **shape** (`type`, `data`, `meta` keys) is defined by `MessageSerializer` regardless of the encoder; the encoder only converts between the raw string and an associative array (JSON ↔ array by default).
```

External producer then always publishes `"type": "file-download"`.

## 2. JSON payload format (JsonMessageEncoder)

> `Yiisoft\Queue\Message\Serializer\MessageSerializer` expects the decoded payload to be an object with these keys (with the default `JsonMessageEncoder`, the wire format is JSON):

Reviewer: Same "wire format". Not clear for non-natives.

Author: fixed
